javalib/src/main/scala/java/nio/channels/FileChannelImpl.scala

```diff
@@ -144,18 +144,21 @@ private object FileChannelImpl {
     val mode = new StringBuilder("r")
     if (options.contains(WRITE) || options.contains(APPEND)) mode.append("w")
 
-    if (!Files.exists(path, Array.empty) && !options.contains(CREATE) && !options
-          .contains(CREATE_NEW)) {
-      throw new NoSuchFileException(path.toString)
-    }
-
-    if (options.contains(CREATE_NEW) && Files.exists(path, Array.empty)) {
-      throw new FileAlreadyExistsException(path.toString)
-    }
-
-    if (options.contains(WRITE) && (options.contains(CREATE_NEW) || options
-          .contains(CREATE))) {
-      Files.createFile(path, attrs)
+    if (!Files.exists(path, Array.empty)) {
+      if (!options.contains(CREATE) && !options.contains(CREATE_NEW)) {
+        throw new NoSuchFileException(path.toString)
+      } else if (options.contains(WRITE)) {
+        Files.createFile(path, attrs)
+      }
```

Comment:
According to the modified version of the  code above you're going to create a new file only if all of the conditions below are met:

1. It it doesn't exist
2. Options doesn't contain CREATE and CREATE_NEW
3. Options contains WRITE

This doesn't seem to match the spec for CREATE and CREATE_NEW that should create new file no matter if WRITE is present or not as defined in https://docs.oracle.com/javase/8/docs/api/java/nio/file/Files.html#newByteChannel-java.nio.file.Path-java.util.Set-java.nio.file.attribute.FileAttribute...-